tests/30rooms/12thirdpartyinvite.pl

```diff
@@ -0,0 +1,419 @@
+use Crypt::NaCl::Sodium;
+use Protocol::Matrix qw( encode_json_for_signing encode_base64_unpadded );
+
+my $crypto_sign = Crypt::NaCl::Sodium->sign;
+
+test "Can invite existing 3pid",
+   requires => [qw( user more_users test_http_server_hostandport )],
+
+   do => sub {
+      my ( $inviter, $more_users, $id_server ) = @_;
+
+      my $invitee_email = "marmosets\@monkeyworld.org";
+      my $invitee_mxid = $more_users->[0]->user_id;
+      my $room_id;
+
+      Future->needs_all(
+         stub_is_lookup( $invitee_email, $invitee_mxid ),
+
+         matrix_create_and_join_room( [ $inviter ], visibility => "private" )
+         ->then( sub {
+            ( $room_id ) = @_;
+            do_request_json_for( $inviter,
+               method => "POST",
+               uri    => "/api/v1/rooms/$room_id/invite",
+
+               content => {
+                  id_server => $id_server,
+                  medium => "email",
+                  address => $invitee_email,
+                  display_name => "Cute things",
+               },
+            ),
```

Comment:
This comma is misleading; it's the end of the function call statement started on line 22. Either remove it, or replace it with `;`.